DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/vrrp_utils.py

```diff
@@ -0,0 +1,155 @@
+            L1 (port/bridge): ip address 192.168.1.4/24
+    """
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [
+        DeviceType.AGGREGATION_ROUTER,
+        DeviceType.INFRA_SWITCH,
+    ], 0)
+    if not tgen_dev or not dent_devices or len(dent_devices) < 3:
+        pytest.skip('The testbed does not have enough devices (1 agg + 2 infra)')
+
+    infra = [dent for dent in dent_devices if dent.type == DeviceType.INFRA_SWITCH]
+    if len(infra) < 2:
+        pytest.skip('The testbed does not have enough infra devices')
+    infra = infra[:2]
+
+    agg = [dent for dent in dent_devices if dent.type == DeviceType.AGGREGATION_ROUTER]
+    if len(agg) < 1:
+        pytest.skip('The testbed does not have enough agg devices')
+    agg = agg[0]
+
+    bridge = 'br0'
+    ep_ip = '192.168.1.5/24'
+    links = [
+        # L0
+        {infra[0]: infra[0].links_dict[agg.host_name][0][0],
+         agg: infra[0].links_dict[agg.host_name][1][0]},
+        # L1
+        {infra[1]: infra[1].links_dict[agg.host_name][0][0],
+         agg: infra[1].links_dict[agg.host_name][1][0]},
+    ]
+
+    # 1. Configure aggregation router
+    out = await IpLink.add(input_data=[{agg.host_name: [
+        {'name': bridge, 'type': 'bridge'}
+    ]}])
+    assert all(res[host_name]['rc'] == 0 for res in out for host_name in res), \
+        'Failed to create bridge'
+
+    out = await IpLink.set(input_data=[{
+        agg.host_name: [
+            {'device': port, 'master': bridge, 'operstate': 'up'}
+            for port in [links[0][agg], links[1][agg]]
+        ] + [
+            {'device': bridge, 'operstate': 'up'}
+        ]
+    }])
+    assert all(res[host_name]['rc'] == 0 for res in out for host_name in res), \
+        'Failed to enslave ports'
+
+    # 2. Configure infra devices
+    if use_bridge:
+        out = await IpLink.add(input_data=[{
+            dent.host_name: [{'name': bridge, 'type': 'bridge'}]
+            for dent in infra
+        }])
+        assert all(res[host_name]['rc'] == 0 for res in out for host_name in res), \
+            'Failed to create bridge'
+
+        out = await IpLink.set(input_data=[{
+            infra[0].host_name: [
+                {'device': links[0][infra[0]], 'operstate': 'up', 'master': bridge},
+                {'device': bridge, 'operstate': 'up'},
+            ],
+            infra[1].host_name: [
+                {'device': links[1][infra[1]], 'operstate': 'up', 'master': bridge},
+                {'device': bridge, 'operstate': 'up'},
+            ],
+        }])
+        assert all(res[host_name]['rc'] == 0 for res in out for host_name in res), \
+            'Failed to enslave ports'
+
+        out = await IpAddress.add(input_data=[{
+            infra[0].host_name: [{'dev': bridge, 'prefix': '192.168.1.3/24'}],
+            infra[1].host_name: [{'dev': bridge, 'prefix': '192.168.1.4/24'}],
+            agg.host_name: [{'dev': bridge, 'prefix': ep_ip}],
```

Comment:
`ipaddress.IPv4Network` could be used to get ip addresses in same network

Reply:
I think this would be overkill. I only need 2 ip addrs, and they are always the same.